tests/fixtures/descriptor_tests/with_create_descriptor_hook/core/hooks/bootstrap.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+# This file is included in the sphinx documentation as an example. Keep this in mind
+# when editing it.
+
+# Everything after this line will be part of the documentation. #documentationStart
+from sgtk import get_hook_baseclass
+from sgtk.util.shotgun import download_and_unpack_attachment
+
+
+class BootstrapHook(get_hook_baseclass()):
+    """
+    This hook allows to download certain bundles from Shotgun instead of
+    official source for the bundle.
+    """
+    def can_cache_bundle(self, descriptor):
+        """
+        Returns true if the descriptor has been cached in Shotgun.
```

Comment:
like i was saying yesterday - my only concern with this more clever approach is that we now have the same docstrings in two places. I worry that it's easy to forget to update this - and perhaps somewhat unintuitive - that we have documentation pulled from the unit tests folder.

Please consider instead have the sphinx docs pulled from the core hook - we could leave the implementation in the hook itself, commented out, users could copy the hook and uncomment that code. We won't get the syntax highlighting in the docs - i am not sure that's a big deal - maybe we can rework the docs to say - here's what the hook looks like, you can copy it, uncomment that stuff and you'll be up and running.

it doesn't seem right to me that we have documentation inside the tests folder.